### ISSUE TITLE
Unobtrusive compiler pass

### ISSUE THREAD

**Issue by EmanueleMinotto** [elcodi/elcodi#992]
The compiler passes currently defined don't pass the [Standard test for unobtrusiveness](https://github.com/matthiasnoback/SymfonyDependencyInjectionTest#standard-test-for-unobtrusiveness).

For example the bundles can't be used with other ORMs like Propel.
Should we wrap these "action" in a condition checking services existence (like in the `MenuChangerCompilerPass`)?

**Comment by mmoreram** [comment]
+1! Great :)

**Comment by mmoreram** [comment]
Just a question... are you talking about DoctrineSimpleMapping definitions?

**Comment by EmanueleMinotto** [comment]
exactly, I think that passes should always act like this: https://github.com/elcodi/elcodi/blob/d65e9ac2c832307a9d6b305fbba38f175cf31378/src/Elcodi/Bundle/MenuBundle/CompilerPass/MenuBuilderCompilerPass.php#L42-L44

**Comment by mmoreram** [comment]
I thought about this a lot, how to decouple from DoctrineSimpleMapping.
Right now is coupled, but I would love to create this uncoupling...
Do you think that right now is the right moment to do that?

[Editing the issue title]

**Comment by EmanueleMinotto** [comment]
Yes, the sooner we decouple it from elcodi, the sooner we'll be able to offer an integration for doctrine ODM.

**Comment by mmoreram** [comment]
Agree!

**Comment by gerfigna** [comment]
+1
